packages/apsara-ui/src/Button/Button.types.tsx

```diff
@@ -1,8 +1,61 @@
-import { ButtonProps } from "antd/lib/button";
 import * as React from "react";
-import { TooltipPlacement } from "antd/lib/tooltip";
 import { CustomIconProps, IconName } from "../Icon/Icon";
 
+const ButtonTypes = ["default", "primary", "ghost", "dashed", "link", "text"] as const;
+export type ButtonType = typeof ButtonTypes[number];
+const ButtonShapes = ["default", "circle", "round"] as const;
+export type ButtonShape = typeof ButtonShapes[number];
+export type SizeType = "small" | "middle" | "large" | undefined;
+const ButtonHTMLTypes = ["submit", "button", "reset"] as const;
+export type ButtonHTMLType = typeof ButtonHTMLTypes[number];
+export interface BaseButtonProps {
+    type?: ButtonType;
+    icon?: React.ReactNode;
+    /**
+     * Shape of Button
+     *
+     * @default default
+     */
+    shape?: ButtonShape;
+    size?: SizeType;
+    disabled?: boolean;
+    loading?: boolean | { delay?: number };
+    prefixCls?: string;
+    className?: string;
+    ghost?: boolean;
+    danger?: boolean;
+    block?: boolean;
+    children?: React.ReactNode;
+}
+export type AnchorButtonProps = {
+    href: string;
+    target?: string;
+    onClick?: React.MouseEventHandler<HTMLAnchorElement>;
+} & BaseButtonProps &
+    Omit<React.AnchorHTMLAttributes<any>, "type" | "onClick">;
+
+export type NativeButtonProps = {
+    htmlType?: ButtonHTMLType;
+    onClick?: React.MouseEventHandler<HTMLButtonElement>;
+} & BaseButtonProps &
+    Omit<React.ButtonHTMLAttributes<any>, "type" | "onClick">;
+
+export type ButtonProps = Partial<AnchorButtonProps & NativeButtonProps>;
+
+export type TooltipPlacement =
+    | "top"
+    | "left"
+    | "right"
+    | "bottom"
+    | "topLeft"
+    | "topRight"
+    | "bottomLeft"
+    | "bottomRight"
+    | "leftTop"
+    | "leftBottom"
+    | "rightTop"
+    | "rightBottom";
```

Comment:
Just a question, are these all antd props?

Reply:
Yes, ButtonProps and TooltipPlacement types were being imported from antd here.